### PR TITLE
fix: resolve flutter static analysis issues (beta)

### LINKED ISSUE #847: Clear up `info` messages from `flutter analyze`
### Which at_widget package you find issue with?

at_backupkey_flutter, at_chat_flutter, at_common_flutter, at_contacts_flutter, at_contacts_group_flutter, at_events_flutter, at_follows_flutter, at_invitation_flutter, at_notify_flutter, at_onboarding_flutter, at_sync_ui_flutter, at_theme_flutter

### On what platforms did you experience the issue?

Linux

### What version are you using?

Latest

### What happened?

In #846 we made the call to enable infos on beta channel so we get early warning of deprecation. This ticket is to clear down existing tech debt.

### Steps to reproduce

```bash
1. Go to any of the listed packages
2. Run `flutter analyze`
```

### Additional info

#842 and #843 take priority, as they're already showing issues in stable channel. This should help us avoid similar problems in the future.

### Flutter analyze output

An example:

```console
Analyzing at_backupkey_flutter...                               

   info • 'background' is deprecated and shouldn't be used. Use surface instead. This feature was deprecated after v3.18.0-0.1.pre • example/lib/main.dart:57:19 • deprecated_member_use
   info • 'background' is deprecated and shouldn't be used. Use surface instead. This feature was deprecated after v3.18.0-0.1.pre • example/lib/main.dart:66:19 • deprecated_member_use
```
```

### PR DESCRIPTION
**Pull request by srieteja**
Closes https://github.com/atsign-foundation/at_widgets/issues/847
**- What I did**
- resolve all issues identified by flutter static analysis on beta channel
- Android:
  1. Upgraded all example apps compileSdkVersion and targetSdkVersion to 34
  2. Set android:exported to true on all example app Manifest.xml (apps do not build without this change)

**- How to verify it**
1. All github actions should pass
2. Manual testing completed

**- Description for the changelog**

fix: resolve flutter static analysis issues (beta)

### CODE REVIEW

**Comment by cpswan** [comment]
@srieteja I've converted this to draft until you can confirm that testing has happened. Do you need help from @CurtlyCritchlow ?

It's good to see the automated tests all green :)

**Comment by srieteja** [comment]
I would appreciate the help, if @CurtlyCritchlow does have bandwidth available.

**Comment by CurtlyCritchlow** [comment]
@srieteja Let's check off the packages as we finish manually testing.
- [x] at_backupkey_flutter
- [x] at_chat_flutter
- [x] at_common_flutter
- [x] at_contacts_flutter
- [x] at_contacts_group_flutter
- [x] at_events_flutter
- [x] at_location_flutter
- [x] at_follows_flutter
- [x] at_invitation_flutter
- [x] at_notify_flutter
- [x] at_onboarding_flutter
- [x] at_sync_ui_flutter
- [x] at_theme_flutter

**Comment by srieteja** [comment]
I've tested the example apps of **at_chat_flutter**, **at_contacts_flutter** and **at_contacts_group_flutter**. All of these show only an onboarding_screen. (I was expecting to see chat/contact widgets on the homepage. Actual widgets might be displayed after onboarding)

Issues Observed (on Android)
1. at_chat_flutter: when I try to onboard/create a new atsign it just gets stuck on a loading screen
2. at_contacts_flutter, at_contacts_group_flutter: when I try to fetch a new atsign, it shows an error dialogue "Unauthorized" (Possibly a missing API key or an old one).
3. There are compatibility issues as the current apps are on 'compileSdkVersion 33', but during build, I was prompted to upgrade to 'compileSdkVersion 34'. Without this change, the build process fails (Not sure if this was something related to my Android SDK setup or current Android platform requirement)

Note: Since these are onboarding issues, the root cause might be lying somewhere in **at_onboarding_flutter**

Since this PR only contains changes to display elements, likely, the changes in this PR do not cause these issues. But we need a separate ticket to address these issues (@CurtlyCritchlow and I will discuss and create a new ticket for this)

**Comment by srieteja** [comment]
showcaseview has released a new major version '3.0.0' during the weekend. I will go ahead and upgrade this dependency for all the packages

**Comment by CurtlyCritchlow** [comment]
> showcaseview has released a new major version '3.0.0' during the weekend. I will go ahead and upgrade this dependency for all the packages

@srieteja, You have to update it in at_backup_flutter. Once that package is updated and published, the issue will be resolved in all the other packages.

**Comment by srieteja** [comment]
> > showcaseview has released a new major version '3.0.0' during the weekend. I will go ahead and upgrade this dependency for all the packages
> 
> @srieteja, You have to update it in at_backup_flutter. Once that package is updated and published, the issue will be resolved in all the other packages.

Yes @CurtlyCritchlow. I have upgraded it in at_backup_flutter, I'm testing all the other packages using a dep_override for at_backup_flutter

**Comment by srieteja** [comment]
> I've tested the example apps of **at_chat_flutter**, **at_contacts_flutter** and **at_contacts_group_flutter**. All of these show only an onboarding_screen. (I was expecting to see chat/contact widgets on the homepage. Actual widgets might be displayed after onboarding)
> 
> Issues Observed (on Android)
> 
> 1. at_chat_flutter: when I try to onboard/create a new atsign it just gets stuck on a loading screen
> 2. at_contacts_flutter, at_contacts_group_flutter: when I try to fetch a new atsign, it shows an error dialogue "Unauthorized" (Possibly a missing API key or an old one).
> 3. There are compatibility issues as the current apps are on 'compileSdkVersion 33', but during the build, I was prompted to upgrade to 'compileSdkVersion 34'. Without this change, the build process fails (Not sure if this was something related to my Android SDK setup or current Android platform requirement)
> 
> Note: Since these are onboarding issues, the root cause might be lying somewhere in **at_onboarding_flutter**
> 
> Since this PR only contains changes to display elements, likely, the changes in this PR do not cause these issues. But we need a separate ticket to address these issues (@CurtlyCritchlow and I will discuss and create a new ticket for this)

### Update:
- I was able to run **at_chat_flutter**, **at_contacts_flutter** and **at_contacts_group_flutter** successfully, when using an existing atsign. The problem I faced earlier was because I was trying to create a new atsign through the example widget. 
- Now I can certainly say that there is an issue with at_onboarding_flutter while trying to create a new atsign (will attach the logs at the end)
- _**reg the issue mentioned with compileSdkVersion**_: It's a platform-related issue and from the articles/tickets I've read it's not a serious issue. This is caused if any of our dependencies have migrated to **compileSdkVersion 34** and we have not. _'It's a standard practice to update/increase the 'compileSdkVersion' occasionally'_ 
**_Ref_**: https://github.com/flutter/flutter/issues/143605

- [Resolved] Faced a new issue with **at_location_flutter**: I could successfully auth with my atKeys and see the at_location UI, but then the app threw an error-dialog saying 'You are not authorized'. I'm not sure what's causing this, but I'm currently working on looking into this. [EDIT] This error was due to the MAP-APIs API key not being set in the example app.

#### _Problematic Logs while creating a new atsign using at_onboarding_flutter_:
`will be updated`

**Comment by CurtlyCritchlow** [comment]
@srieteja all the apps work for me. It's a green light for me.

File: packages/at_theme_flutter/example/pubspec.yaml [hunk 1]
```diff
@@ -39,6 +39,12 @@ dependencies:
   flutter:
     sdk: flutter
 
+dependency_overrides:
```
Comment (cpswan) [inline packages/at_theme_flutter/example/pubspec.yaml:42]: Should we still have a dependency override here now that showcaseview 3.0.0 is available?

**Review by cpswan — CHANGES_REQUESTED** [review]

**Comment by cpswan** [comment]
Great work @srieteja and @CurtlyCritchlow

There's a lot going on here, and I think it's 99% good to go. Just one niggle about a(n overlooked?) dependency override.

File: packages/at_theme_flutter/example/pubspec.yaml (hunk 1 above)
Comment (srieteja) [reply packages/at_theme_flutter/example/pubspec.yaml:42]: That has been fixed in [6be04c8](https://github.com/atsign-foundation/at_widgets/pull/852/commits/6be04c8ecc2247a724648a05098f87567c9394eb). I'm using a local dependency for now, which will be changed after publishing a new version of at_backupkey_flutter.

**Review by cpswan — APPROVED** [review]

**Comment by cpswan** [comment]
@srieteja are you going to take care of publishing these to pub.dev now?

**Comment by srieteja** [comment]
@cpswan, I will publish at_backupkey_flutter and at_onboarding_flutter later today, as these are key packages. @CurtlyCritchlow and I can coordinate on publishing the rest.

**Pull request merged**